### ISSUE TITLE
BUG: on android, excalidraw canvas don't follow keyboard hiding

### ISSUE THREAD

**Issue by JE-Amouel** [zsviczian/obsidian-excalidraw-plugin#1729]
**Your environment**
Android 12, on Tecno pop 8

**Describe the bug**
When I hide the virtual keyboard with the arrow buttons on the bottom left of the screen, excalidraw don't sence the effective change, so it doesn't restore canvas in full screen

**To Reproduce**
Entre obsidian
Open excalidraw canvas
Use text tools, 
Choose a place to enter text
When the virtual keyboard appears
Hide it with the hiding button on the bottom left of the screen,

**Expected behavior**
When the keyboard appears, excalidraw adjust the canvas dimension as per the keyboard dimension, so after hiding the virtual keyboard it should restore it dimension. 

**videos**

https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/146543282/42088ed5-b2ce-4f42-b03b-4415d822e365

**Comment by zsviczian** [comment]
yes - on-screen keyboards are a pain in the... 
The current solution is not 100% perfect, but frankly, I am happy it works at least this well. 
I have an S23 Ultra. I can't reproduce the issue, but I can understand why it may be happening. 
I would expect this does not always happen. Does it?

**Comment by zsviczian** [comment]
I see, this only happens if you start to edit text, but then enter no text and hide the keyboard.... I am able to reproduce. Let me take a look

**Comment by zsviczian** [comment]
I hope to have resolved this in 2.1.7
Please test it once released and let me know.
It will be another few days before I release 2.1.7...

**Comment by JE-Amouel** [comment]
Thank you.

**Comment by JE-Amouel** [comment]
Thanks a lot, that work fine, and sorry for the delay.

**Comment by Dnarts** [comment]
@zsviczian same problem. 
nothing phone 3a pro (android 15)
😕

**Comment by zsviczian** [comment]
Try a different keyboard app. In some cases we found that the way the keyboard app builds the keyboard affects how Excalidraw can notice the change. Unfortunately, the CSS page environment provided by Obsidian, and the lack of a keyboard event on mobile devices I can only implement heuristics to handle on-screen popup. Depending on the keyboard you are using (and also the performance of your device) this can be an issue.

**Comment by Dnarts** [comment]
> Try a different keyboard app. 

I found solution the trick was to change display cutout (maybe also dpi)  in developers system settings of the phone,  only this one of them reacted positively and fixed that bug

<img width="1080" height="2392" alt="Image" src="https://github.com/user-attachments/assets/e5cbe21f-f996-4ad1-96ad-f24a3ab55328" />

**Comment by Dnarts** [comment]
Oh that was very tricky option andin the end I just installed the new one. And It's perfectly Fix the bug in obsidian and in all other apps. Because it's floating keyboard, just look at this miracle.

<img width="756" height="1674" alt="Image" src="https://github.com/user-attachments/assets/351a67ed-5cb9-4e82-9cdf-198bbe689d39" />

I'm advice to everyone to try it. Amazing.
It's called [Yandex keyboard.](https://play.google.com/store/apps/details?id=ru.yandex.androidkeyboard)

**Comment by zsviczian** [comment]
yeah, floating keyboards are a much better choice with Obsidian-Excalidraw.
